src/site/content/en/blog/ptz/index.md

```diff
@@ -0,0 +1,230 @@
+  <figcaption class="w-figcaption">Internal page to debug PTZ camera support.</figcaption>
+</figure>
+
+### Control camera PTZ {: #control }
+
+Manipulate camera PTZ capabilities and settings using the preview
+`MediaStreamTrack` from the `stream` object obtained earlier.
+`MediaStreamTrack.getCapabilities()` returns a dictionary with the supported
+capabilities and the ranges or allowed values. Correspondingly,
+`MediaStreamTrack.getSettings()` returns the current settings.
+
+Pan, tilt, and zoom capabilities and settings are available only if supported by
+the camera and the user has granted PTZ permission to the camera.
+
+<figure class="w-figure">
+  <video controls autoplay loop muted class="w-screenshot">
+    <source src="https://storage.googleapis.com/web-dev-assets/ptz/ptz_h264.mp4" type="video/mp4">
+  </video>
+  <figcaption class="w-figcaption">Controlling Camera PTZ.</figcaption>
+</figure>
+
+Call `videoTrack.applyConstraints()` with the appropriate [PTZ advanced
+constraints] to control camera pan, tilt, and zoom as shown in the example below.
+The returned promise will resolve if successful. Otherwise it will reject if
+either:
+- the camera with PTZ permission is not granted.
+- the camera hardware does not support the PTZ constraint.
+- the page is not visible to the user. Use the [Page Visibilty API] to detect
+  page visibility changes.
+
+```js
+// Get video track capabilities and settings.
+const [videoTrack] = stream.getVideoTracks();
+const capabilities = videoTrack.getCapabilities();
+const settings = videoTrack.getSettings();
+
+// Let the user control the camera pan motion if the camera supports it
+// and PTZ access is granted.
+if ("pan" in settings) {
+  const input = document.querySelector("input[type=range]");
+  input.min = capabilities.pan.min;
+  input.max = capabilities.pan.max;
+  input.step = capabilities.pan.step;
+  input.value = settings.pan;
+
+  input.addEventListener("input", async () => {
+    await videoTrack.applyConstraints({ advanced: [{ pan: input.value }] });
+  });
+}
+
+if ("tilt" in settings) {
+  // similar for tilt...
+}
+if ("zoom" in settings) {
+  // similar for zoom...
+}
+```
+
+It is also possible to configure camera pan, tilt, and zoom by calling
+`navigator.mediaDevices.getUserMedia()` with some camera PTZ ideal constraint
+values. This is handy when camera PTZ capabilities are known in advance. Note
+that [mandatory constraints] (min, max, exact) are not allowed here.
+
+```js
+const stream = await navigator.mediaDevices.getUserMedia({
+  // Website asks to reset known camera pan.
+  video: { pan: 0, deviceId: { exact: "myCameraDeviceId" } }
+});
+```
+
+## Playground  {: #playground }
+
+You can play with the API by running the [demo] on Glitch. Be sure to [check out
+the source code].
+
+Tip: If you don't have a camera that supports PTZ, you can [run Chrome with the
+switch] `--use-fake-device-for-media-stream` to simulate one on your machine.
+Enjoy!
+
+{% Glitch {
```

Comment:
You make use of `color-scheme: dark light` (🎉), but the iframe doesn't respect it, only the standalone doc. You need to add…

```css
@media (prefers-color-scheme: dark) {
  body {
    color: #fff; 
    background-color: #000;
  }
}
```

…to your CSS.

Reply:
As discussed offline, this is not enough. I think we should add proper dark theme support to all web.dev pages.